ktor-network/jvm/src/io/ktor/network/sockets/CIOReader.kt

```diff
@@ -26,19 +27,24 @@ internal fun CoroutineScope.attachForReadingImpl(
     val buffer = pool.borrow()
     return writer(Dispatchers.Unconfined + CoroutineName("cio-from-nio-reader"), channel) {
         try {
+            val timeout = if (socketOptions?.socketTimeout != null) {
```

Comment:
Let's discuss why we need this coroutine at all

Reply:
discussed offline